src/pages/resources.js

```diff
@@ -16,12 +16,45 @@ const ResourcePage = () => {
       !!category.resources?.nodes || !!category.resources?.nodes?.length
   );
 
+  const [searchQuery, setSearchQuery] = useState("");
+  const [searchResult, setSearchResult] = useState(allResourcesCategory);
+
+  useEffect(() => {
+    if (searchQuery === "") {
+      setSearchResult(allResourcesCategory);
+    } else {
+      const lowercaseSearchQuery = searchQuery.toLowerCase().trim();
+
+      const filteredResult = allResourcesCategory
+        .filter(category =>
+          category.resources.nodes.some(node =>
+            node.title.toLowerCase().includes(lowercaseSearchQuery)
+          )
+        )
+        .map(category => {
+          category.resources.nodes = category.resources.nodes.filter(node =>
+            node.title.toLowerCase().includes(lowercaseSearchQuery)
+          );
+
+          return category;
+        });
+
+      setSearchResult(filteredResult);
+    }
+
+    return () => {
+      setSearchResult(allResourcesCategory);
+    };
+  }, [searchQuery]);
```

Comment:
> This `useEffect` is not needed.
> 
> The component would `re-render` everytime the `searchQuery` value is updated as it stands. The logic for checking for empty search value can go in the filter.
> 
> `useEffect` by design is used for synchronizing external data which is not the case here. cc @Joponpin2000 @Demoivre02

> I think based on what he wants to accomplish, `useEffect` might just be needed, i think the aim is to update the `resourceCategory` on every letter input. I think one solution is to make the seach bar independent, or to garb the userinput as a whole, then query it with the `resourceCategory` with an onclick function.

Reply:
Nope, still not needed, the component already re-renders with the the `useState`, no need to re-render again with `useEffect`, you mostly want to use `useEffect` when you are syncing with `external` state say an API. Here is an helpful resource from the React Documentation: https://react.dev/learn/you-might-not-need-an-effect . Any logic needed can go in the filter function and it would still work fine. cc @Demoivre02 @Joponpin2000